src/main/java/core/basesyntax/service/operation/BalanceOperation.java

```diff
@@ -0,0 +1,20 @@
+package core.basesyntax.service.operation;
+
+import core.basesyntax.db.FruitStorage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.model.FruitsForSale;
+
+public class BalanceOperation implements OperationHandler {
+
+    @Override
+    public void updateDataBase(FruitTransaction transaction) {
+        FruitsForSale fruitsForSale = new FruitsForSale();
+        if (fruitsForSale.isNotForSale(transaction.getFruit())) {
+            throw new RuntimeException("We don't sell such fruit: " + transaction.getFruit());
+        }
+        if (transaction.getQuantity() < 0) {
+            throw new RuntimeException("The balance cannot be negative.");
```

Comment:
The error message could be more informative by including the fruit name and the attempted negative quantity. This would help in debugging and understanding which transaction caused the error.